### ISSUE TITLE
Namespace selection by labels in roles `kubernetes_resources`

### ISSUE THREAD

**Issue by Caomoji** [gravitational/teleport#45635]
What would you like Teleport to do?

It would be useful to be able to select namespaces in Teleport roles `spec.allow.kubernetes_resources` by label rather than only their name.

What problem does this solve?

It would allow to provide better filtering on namespaces that have names difficult to include as part of an automation system. For instance, if we have namespaces labeled with their team, we could use this simple selector to list all namespaces the team's role should allow, rather than having to know, in Teleport configuration, which namespaces are owned by which team.

**Comment by zmb3** [comment]
@rosstimothy good to close this one?

**Comment by rosstimothy** [comment]
> @rosstimothy good to close this one?

No. We don't currently support selecting by labels of Kubernetes resources in `kubernetes_resources`.
